sklearn/cluster/tests/test_optics.py

```diff
@@ -95,27 +95,18 @@ def test_dbscan_optics_parity():
     op = OPTICS(min_samples=10).fit(X)
     core_optics, labels_optics = op.extract_dbscan(0.3)
 
-    op_h = metrics.homogeneity_score(labels_true, labels_optics)
-    op_c = metrics.completeness_score(labels_true, labels_optics)
-    op_V = metrics.v_measure_score(labels_true, labels_optics)
-    op_m = metrics.adjusted_mutual_info_score(labels_true, labels_optics)
-    op_s = metrics.silhouette_score(X, labels_optics)
-
-    op_metrics = np.array([op_h, op_c, op_V, op_m, op_s])
-
-    # calculate dbscan labels and cluster metrics
+    # calculate dbscan labels
     db = DBSCAN(eps=0.3, min_samples=10).fit(X)
 
-    db_h = metrics.homogeneity_score(labels_true, db.labels_)
-    db_c = metrics.completeness_score(labels_true, db.labels_)
-    db_V = metrics.v_measure_score(labels_true, db.labels_)
-    db_m = metrics.adjusted_mutual_info_score(labels_true, db.labels_)
-    db_s = metrics.silhouette_score(X, db.labels_)
+    confusion_m = contingency_matrix(db.labels_, labels_optics)
```

Comment:
I'd rather `contingency` to reduce confusion: a confusion matrix relies on having a fixed identity of categories, while contingency is a more general idea that can account for arbitrary cluster IDs.

Reply:
ok, fixed.